### ISSUE TITLE
WCObject constructor with custom object fails

### ISSUE THREAD

**Issue by vankampenp** [XiaoFaye/WooCommerce.NET#247]
WooCommerce.NET 0.7.4

When creating a new WCObject with custom objects:
 ```
WCObject<Coupon, MpCustomer, Product, ProductReview, Variation, MpOrder, OrderNote, OrderRefund, ProductAttribute, ProductAttributeTerm, ProductCategory, ShippingClass, ProductTag, TaxRate, TaxClass>
                    wc = new WCObject<Coupon, MpCustomer, Product, ProductReview, Variation, mp_Order, OrderNote, OrderRefund, ProductAttribute, ProductAttributeTerm, ProductCategory, ShippingClass, ProductTag, TaxRate, TaxClass>
                        (rest);
```
The assignment in WCObject.cs line 26 fails

`Customer = new WCItem<T2>(api);
`

This because in the WCItem constructor (BaseObjects.cs line 129)
fails with the assigment
`APIEndpoint = typeof(T).GetRuntimeProperty("Endpoint").GetValue(null).ToString();`

This fails because 
`typeof(T).GetRuntimeProperty("Endpoint")`
returns null when the class is derived:
`[DataContract]
    public class MpCustomer : Customer`

As a workaround, I have now supplied the EndPoint also in my derived class:

```
  [DataContract]
    public class MpCustomer : Customer
    {
          public new static string Endpoint { get; } = @"customers";
....
}
```
With that "customers" is assigned to the APIEndpoint

**Comment by XiaoFaye** [comment]
Can you please try to change below:
`typeof(T).GetRuntimeProperty("Endpoint").GetValue(null).ToString();`
to
```
            if (typeof(T).BaseType == typeof(JsonObject))
                APIEndpoint = typeof(T).GetRuntimeProperty("Endpoint").GetValue(null).ToString();
            else
                APIEndpoint = typeof(T).BaseType.GetRuntimeProperty("Endpoint").GetValue(null).ToString();
```
See if this will work in both scenarios.

**Comment by XiaoFaye** [comment]
@vankampenp  any luck on this?

**Comment by vankampenp** [comment]
Hi, thanks for the tip.
The code does not work with the classes that are not based on JsonObject, such as WCAttributeItem.

This does work:
             if (typeof(T).Namespace.StartsWith("WooCommerceNET.WooCommerce"))
                APIEndpoint = typeof(T).GetRuntimeProperty("Endpoint").GetValue(null).ToString();
            else
                APIEndpoint = typeof(T).BaseType.GetRuntimeProperty("Endpoint").GetValue(null).ToString();

But I don't know if you can be sure that someone is not going to define their custom classes in the WooCommerceNet.WooCommerce namespace.

**Comment by XiaoFaye** [comment]
this is fixed in version 0.7.4
